libres/lib/analysis/update.cpp

```diff
@@ -670,4 +619,87 @@ static bool smoother_update(py::object updatestep, int total_ens_size,
+                                   std::vector<bool> ens_mask) {
+    auto ensemble_config_ =
+        ert::from_cwrap<ensemble_config_type>(ensemble_config);
+    auto source_fs_ = ert::from_cwrap<enkf_fs_type>(source_fs);
+    auto target_fs_ = ert::from_cwrap<enkf_fs_type>(target_fs);
+    return analysis::copy_parameters(source_fs_, target_fs_, ensemble_config_,
+                                     ens_mask);
+}
+
+std::shared_ptr<analysis::update_data_type>
+make_update_data_pybind(py::object source_fs, py::object target_fs,
+                        py::object obs, py::object ensemble_config,
+                        py::object analysis_config, std::vector<bool> ens_mask,
+                        py::object ministep, py::object shared_rng) {
+    auto source_fs_ = ert::from_cwrap<enkf_fs_type>(source_fs);
+    auto target_fs_ = ert::from_cwrap<enkf_fs_type>(target_fs);
+    auto obs_ = ert::from_cwrap<enkf_obs_type>(obs);
+    auto ensemble_config_ =
+        ert::from_cwrap<ensemble_config_type>(ensemble_config);
+    auto analysis_config_ =
+        ert::from_cwrap<analysis_config_type>(analysis_config);
+    auto ministep_ = ert::from_cwrap<local_ministep_type>(ministep);
+    auto shared_rng_ = ert::from_cwrap<rng_type>(shared_rng);
+    return analysis::make_update_data(source_fs_, target_fs_, obs_,
+                                      ensemble_config_, analysis_config_,
+                                      ens_mask, ministep_, shared_rng_);
+}
+
+void save_parameters_pybind(py::object target_fs, py::object ensemble_config,
+                            std::vector<int> iens_active_index,
+                            py::object ministep,
+                            analysis::update_data_type &update_data) {
+    auto target_fs_ = ert::from_cwrap<enkf_fs_type>(target_fs);
+    auto ensemble_config_ =
+        ert::from_cwrap<ensemble_config_type>(ensemble_config);
+    auto ministep_ = ert::from_cwrap<local_ministep_type>(ministep);
+    return analysis::save_parameters(target_fs_, ensemble_config_,
+                                     iens_active_index, ministep_, update_data);
+}
+void run_analysis_update_without_rowscaling_pybind(
+    const ies::config::Config &module_config, ies::data::Data &module_data,
+    std::vector<bool> ens_mask, analysis::update_data_type &update_data) {
+
+    analysis::run_analysis_update_without_rowscaling(
+        module_config, module_data, ens_mask, update_data.obs_mask,
+        &update_data.S, &update_data.E, &update_data.D, &update_data.R,
+        &update_data.A.value());
+}
+
+void run_analysis_update_with_rowscaling_pybind(
+    const ies::config::Config &module_config, ies::data::Data &module_data,
+    analysis::update_data_type &update_data) {
+
+    analysis::run_analysis_update_with_rowscaling(
+        module_config, module_data, &update_data.S, &update_data.E,
+        &update_data.D, &update_data.R, update_data.A_with_rowscaling);
+}
```

Comment:
Make these `static` or wrap them in `namespace {` (anonymous namespace). This is so that these functions aren't exposed outside of the `.cpp` file.